### PR TITLE
Handle Download, Share, Add to Cart for Files having special characte…

### PR DESCRIPTION
**Pull request by sandeep-prasad**
The files having special character in the filename, Download, Share and Add to Cart for the files was not successful.
Code Fix: Encoded the paths of the files and the sent it to request parameter and decoded the file paths in the servlet.

### CODE REVIEW

**Comment by kaushalmall** [comment]
hi @davidjgonzalez, mind looking at this? I checked and everything seems to be OK.

**Comment by kaushalmall** [comment]
@sandeep-prasad you will need to create an issue with proper information and then update the changelog to add the issue number.

File: core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/share/impl/EmailShareServiceImpl.java [hunk 1]
```diff
@@ -135,14 +138,23 @@ public final void share(final SlingHttpServletRequest request, final SlingHttpSe
...
+        final String[] assetEncodedPaths = shareParameters.get(ASSET_PATHS, String[].class);
 
         // Check to ensure the minimum set of e-mail parameters are provided; Throw exception if not.
         if (emailAddresses == null || emailAddresses.length == 0) {
             throw new ShareException("At least one e-mail address is required to share");
-        } else if (assetPaths == null || assetPaths.length == 0) {
+        } else if (assetEncodedPaths == null || assetEncodedPaths.length == 0) {
             throw new ShareException("At least one asset is required to share");
         }
+        int arrayLength = assetEncodedPaths.length;
```
Comment (davidjgonzalez) [inline core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/share/impl/EmailShareServiceImpl.java]: This looks rather .. verbose? Can't we do a nice stream() online 141... i think its reasonable to catch encoding exceptions and use the string as-is?

```
final String[] assetPaths = Stream.of(shareParameters.get(ASSET_PATHS, String[]{}))
.map(p -> try { return URLDecoder.decode(p, "UTF-8"); } 
                  catch (UnsupportedEncodingException e)  { log.warn("Could not decode path [ {} ], using as is", p); return p; })
.filter(p -> StringUtils::isBlank)
.toArray();
```
or something?

File: ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/modals/share/share.html [hunk 2]
```diff
@@ -78,7 +78,8 @@ <h3>${properties['errorTitle']}</h3>
 				<div class="ui header">${properties['shareOptionsTitle']}</div>
 	
 				<sly data-sly-list.asset="${share.assets}">
-					<input type="hidden" name="path" value="${asset.path}" />
+				<sly data-sly-use.encodedPath="${'../encodePath.js' @ url=asset.path}"></sly>
+					<input type="hidden" name="path" value="${encodedPath}" />
```
[the remark this answers is not in the document]
Comment (davidjgonzalez) [reply ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/modals/share/share.html]: Should we make a Sling Model (some generic "ComponentUtil" model? .. or we could also make a computed property for the Url? ... that might be cleaner?

**Comment by sandeep-prasad** [comment]
@davidjgonzalez : 
Incorporated the review comments.
1. Removed JS Use file and added a Sling model, CompUtil.java.
2. Updated the Java 8 code snippet for Iterating the array.

Thanks, Please review.

**Comment by sandeep-prasad** [comment]
@davidjgonzalez , @kaushalmall : Hi can we prioritize the pull request.

**Comment by davidjgonzalez** [comment]
Closing - Using #305 which cleans up this PR.

**Pull request closed without merging**